tests/PHPStan/Analyser/nsrt/preg_match_shapes.php

```diff
@@ -7,117 +7,117 @@
 
 function doMatch(string $s): void {
```

Comment:
https://3v4l.org/Oq0GB

some usecases that might be work to be tested

Reply:
Please create a phpstan.org/try snippet with type-asserts

Reply:
You tagged me and this is my answer, I am pointing out "maybe problematic regexes", not exactly the types.